flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/delegation/Planner.java

```diff
@@ -74,14 +76,22 @@
 	 */
 	List<Transformation<?>> translate(List<ModifyOperation> modifyOperations);
 
+	/**
+	 * Creates a {@link SelectTableSink} for a select query.
+	 *
+	 * @param tableSchema the table schema of select result.
+	 * @return The {@link SelectTableSink} for the select query.
+	 */
+	SelectTableSink createSelectTableSink(TableSchema tableSchema);
```

Comment:
This interface looks strange to me. An alternative approach is we create another dedicated `ModifyOperation` which used for collecting, and reuse the `List<Transformation<?>> translate(List<ModifyOperation> modifyOperations);`. We can encapsulate the logic under this method.

Reply:
I think it does not work. because collect Iterator is generated in `TableSink` (e.g. the iterator is created when `StreamSelectTableSink` is created.) and we need to put it into `TableResult` in `TableEnvironmentImpl#executeOperation`. the `SelectTableSink` instance does not change, we can get the Iterator. 

while we can't get collect Iterator from the dedicated `ModifyOperation`.

Reply:
ok, let's keep it for now and revisit this later if we have better idea.